### PR TITLE
docs(input): add missing number input docs

### PR DESCRIPTION
**Pull request by nkrantz**
Add i18n props to props table, add Internationalization section and example, remove outdated Anatomy section

### CODE REVIEW

File: packages/paste-website/src/pages/components/input/index.mdx [hunk 1]
```diff
@@ -239,12 +239,24 @@ Help Text should also include any important contextual information about what th
 
 <LivePreview scope={{Input, Label, HelpText}} language="jsx">
   {`<>
-  <Label htmlFor="reload_amount">Reload amount</Label>
+  <Label htmlFor="reload_amount">Number Input</Label>
```
Comment (TheSisb) [inline packages/paste-website/src/pages/components/input/index.mdx:242]: question: accidental change?

File: packages/paste-website/src/pages/components/input/index.mdx [hunk 2]
```diff
@@ -239,12 +239,24 @@ Help Text should also include any important contextual information about what th
...
   <Input aria-describedby="reload_amount_help" id="reload_amount" name="reload_amount" defaultValue="0" type="number" min="0" max="1000" />
   <HelpText id="reload_amount_help">Maximum amount is $1,000.</HelpText>
 </>`}
 </LivePreview>
 
+#### Internationalization
+
+To internationalize Number Input, provide translated strings to be used for the increment and decrement button accessible labels. Use the `i18nStepUpLabel` and `i18nStepDownLabel` props.
```
Comment (TheSisb) [inline packages/paste-website/src/pages/components/input/index.mdx:250]:
```suggestion
Provide translated strings for the 'increment' and 'decrement' button labels to make NumberInput localized. The `i18nStepUpLabel` and `i18nStepDownLabel` props should be used for this purpose.
```

**Review by TheSisb — APPROVED** [review]

File: packages/paste-website/src/pages/components/input/index.mdx (hunk 2 above)
Comment (TheSisb) [reply packages/paste-website/src/pages/components/input/index.mdx:250]: This was marked as resolved without any changes afaict. It reads a little weird to me, but not gonna block on it.

**Review by SiTaggart — APPROVED** [review]

File: packages/paste-website/src/pages/components/input/index.mdx (hunk 2 above)
Comment (nkrantz) [reply packages/paste-website/src/pages/components/input/index.mdx:250]: weird, I made the changes but they got lost

**Review by TheSisb — APPROVED** [review]

**Review by loreina — APPROVED** [review]

**Pull request merged**